src/term/net_to_term.rs

```diff
@@ -275,52 +275,31 @@ impl<'a> Reader<'a> {
   fn resugar_string(&mut self, term: &mut Term) -> Term {
```

Comment:
Could you refactor these `impl` blocks so that they're more organized? Right now `resugar_string` and `resugar_list` are in a separate `impl` from `resugar_adts` and we have 3 impl blocks for Reader and 2 impl blocks for Term.

Maybe separate this file into sections, each doing one thing, or extract adt resugaring to a separate pass in a different file. 
I think overall this file is a bit disorganized.